File: demo/safira.c
#include <stdio.h>

// Define a structure for Gemstone
struct Gemstone {
    char name[20];
    char color[20];
    char rarity[20];
};

// Function to describe the gemstone
void describeGemstone(struct Gemstone gemstone) {
    printf("The %s gemstone is a beautiful shade of %s and is %s.\n", gemstone.name, gemstone.color, gemstone.rarity);
}

int main() {
    // Create a Gemstone object
    struct Gemstone myGemstone = {"safira", "blue", "rare"};

    // Describe the gemstone
    describeGemstone(myGemstone);

    return 0;
}

File: demo/safira.go
package main

import "fmt"

// Define a struct for Gemstone
type Gemstone struct {
    name   string
    color  string
    rarity string
}

// Method to describe the gemstone
func (g Gemstone) describe() {
    fmt.Printf("The %s gemstone is a beautiful shade of %s and is %s.\n", g.name, g.color, g.rarity)
}

func main() {
    // Create a Gemstone object
    myGemstone := Gemstone{name: "safira", color: "blue", rarity: "rare"}

    // Describe the gemstone
    myGemstone.describe()
}
